### DISCUSSION TITLE
Wikibase Cloud or Wikibase Suite?

### DISCUSSION THREAD

**Discussion by CECSpecialistI** [crystalyragui/MARC2RDA#457] [Q&A]
We have decided to pursue Wikibase as the method of hosting our output data. There are two viable options--Wikibase Cloud and Wikibase Suite. We need to figure out which path to choose. Let's discuss!

**Comment by hinxcode** [comment]
Hi @CECSpecialistI @cspayne ,

I have just successfully set up a local Wikibase Suite on the computer in CAMS. I'm wondering if there are any specific features we want to try first on Wikibase Suite. For instance, it would be helpful if you could provide more details on which catalog we can test, so I can offer further insights, like the pros/cons of Wikibase Suite.

For now, here are some points we might consider if we choose Wikibase Suite over Wikibase Cloud:
1. [Hardware requirements](https://github.com/wmde/wikibase-release-pipeline/tree/main/deploy#hardware), which might involve coordination with UW-IT or other VPS hosting providers.
2. Extra time for self-hosting, including the setup of public IPs, SSL, and collaboration efforts with UW-IT.
3. Potential future issues, such as network congestion for the server and software upgrades.

**Reply by CECSpecialistI** [reply]
Let's first try and see if we can get all the properties we want ingested? Basically we want the entire RDA registry. I think @szapoun & colleagues at the national library of greece wrote a program to ingest all the properties and get them into their Wikibase without needing to upload them manually. For interoperability, it would be neat if we used the same exact properties they are using. I don't know if that is possible. We also need to meet with ITS. We should use the University of Washington catalog for initial testing, probably only records we have created originally in OCLC to avoid any copyright entanglements. I think I can start an export of a set tomorrow. It will be big, which is good that's what we want. Then we will need to run it through the converter, though. And then convert the RDA/RDF properties to our local Wikibase properties. Thank you for doing this work @hinxcode !!

**Comment by CECSpecialistI** [comment]
LD4 has a Wikibase Working Group which has Wikibase working hours. They are recorded [here](https://m.wikidata.org/w/index.php?title=Wikidata:WikiProject_LD4_Wikidata_Affinity_Group/Wikibase_Working_Hours)

**Comment by CECSpecialistI** [comment]
Continue to explore Wikibase Suite and use it for testing/setup. For production, pursue Wikibase Cloud as far as it's possible.

**Reply by hinxcode** [reply]
Regarding Gordon's concern, I agree that crowdsourcing can be highly effective, especially when we align with the values of Wikimedia experts. However, I wonder if the optimizations or solutions we will learn from Enslaved.org (such as how they managed the SPARQL endpoint) can be applied to Wikibase Cloud. This would likely require additional research and testing. If these customizations cannot be migrated, I am concerned that meeting our specific needs could become challenging, as our feature requirements may not be prioritized by others.

**Reply by CECSpecialistI** [reply]
According to Jason, the optimizations/solutions employed by enslaved.org cannot be applied to Wikibase Cloud. We  would be relying on WMDE developers to meet our needs in the Cloud environment, without our own options for optimization. In others' experience, fixes have taken several months but have come through. However, the fixes are free of charge.

**Reply by CECSpecialistI** [reply]
You are right that our feature requirements may not be prioritized by others, particularly when it comes to query timeouts.

**Reply by CECSpecialistI** [reply]
To be clear, crowdsourcing and public participation are not different in Wikibase Suite vs. Wikibase Cloud instances. The person/group maintaining the Wikibase instance (UW either way, Wikibase Cloud will just be hosted on a WMF server) decides whether the Wikibase instance is public or private and whether people can make accounts to edit the database or not. Our database will be set to public so that people can make accounts and edit it in either environment. The choice between Wikibase Cloud and Wikibase Suite has zero impact on this.

**Comment by CECSpecialistI** [comment]
I just met with Jason Evans from National Library of Wales about his experience with Wikibase Cloud. [Notes are here](https://docs.google.com/document/d/1xK0JZOaos5HU-KnogHeo0iJoz00B9IachK8b-vmGIJE/edit?usp=sharing)

**Reply by CECSpecialistI** [reply]
It sounds like our biggest pain point, whether we go with the Cloud or Suite, is going to be the SPARQL endpoint for the query service. Our Wikibase will dwarf the largest Wikibase on the Cloud by a significant  margin. Jason advised that we get in touch with the folks at Enslaved.org about their Wikibase Suite instance and what they have done to improve their SPARQL endpoint. I emailed them today.

**Reply by hinxcode** [reply]
Thank you, Crystal!
These meeting notes are helpful. Two surprising points for me are the data ingestion speed (5,000 items/hour sounds quite slow lol) and the number of active editors (only 1?!). Anyway, I will also take a look and test if WDQS works well on the student computer next week. Looking forward to hearing the update from enslaved.org

**Reply by hinxcode** [reply]
Hi @CECSpecialistI ,
Could you ask them if the Wikibase engineers have scaled up/out of their WB Cloud to handle the high usage and traffic? If they have, could they share technical information on how they accomplished this? For example, how many resources are needed to run their instance. We might need to set up a similar environment for our Wikibase Suite.

**Reply by CECSpecialistI** [reply]
> Hi @CECSpecialistI , Could you ask them if the Wikibase engineers have scaled up/out of their WB Cloud to handle the high usage and traffic? If they have, could they share technical information on how they accomplished this? For example, how many resources are needed to run their instance. We might need to set up a similar environment for our Wikibase Suite.

Whom do you mean here? Do you mean ask Wikimedia Deutschland/Wikibase Cloud folks? Enslaved.org isn't running on Wikibase Cloud.

**Reply by hinxcode** [reply]
Leave a note here. I just discussed this with Crystal. We're gathering [questions](https://docs.google.com/document/d/1mk1Vu7ZjaErBsS1HCHraEO_gu5dgUzP3H34ysoh65xs/edit) for Wikimedia Deutschland to gain more experience from them which would answer my original question.

**Comment by hinxcode** [comment]
The list below is the first draft of the hardware requirements. Please note that this list should be updated as we receive more information about WB ~Cloud~ Suite configurations from other organizations.

## Hardware Requirements List
1. AMD64 architecture multicore processor, minimum 4 cores
2. RAM: Minimum 8 GB
3. Linux distribution 
4. Enough storage for development needs (to be discussed)
5. Access to update DNS records (or making sure we can file tickets to ITS)
6. Stable and reliable network connection
7. Virtual machine management console access (only if ITS provides Virtual Nodes instead of physical machines)
8. Public IP address (can be ignored if we will use Wikibase Cloud for production env)

**Reply by CECSpecialistI** [reply]
Is this for WB Cloud or Wikibase Suite? This looks like WB Suite

**Reply by hinxcode** [reply]
Sorry, I made a typo here. This is for WB Suite.

**Comment by CECSpecialistI** [comment]
@hinxcode can you find out how much it would cost to host a 4TB wikibase suite instance on AWS for 3 years?

**Reply by hinxcode** [reply]
@CECSpecialistI Apologies for the delay.

Based on the results from my testing and experimentation in the CAMS computer's local environment, the estimated AWS cost for 3 years is approximately **$19,527.84** ($542.44 per month). This architecture is designed to support a production environment (for public users) and has been optimized specifically for the SPARQL service to minimize performance issues (custom programs are still needed though).

Please find the detailed estimate attached for more info.
[My Estimate - AWS Pricing Calculator.pdf](https://github.com/user-attachments/files/16355281/My.Estimate.-.AWS.Pricing.Calculator.pdf)

**Reply by hinxcode** [reply]
To maximize cost savings, reducing server performance could save up to **$213** per month, resulting in an annual savings of $213 x 12 = **$2,556**. It's flexible for adjusting on the go. However, we must pay special attention to potential server downtime issues if traffic emerges.

**Comment by CECSpecialistI** [comment]
Here's a [document](https://docs.google.com/document/d/1mk1Vu7ZjaErBsS1HCHraEO_gu5dgUzP3H34ysoh65xs/edit?usp=sharing) for us to track our questions for outside people/groups

**Comment by CECSpecialistI** [comment]
Crystal, Cypress, and Ying-Hsiang just met with UW ITS again. Minutes [here](https://docs.google.com/document/d/1Nhz_gjo9J1W31mhEI-p8sRky8pVxURGhXmSmg1Dw23I/edit?usp=sharing)
